tests/integration.rs

```diff
@@ -5712,6 +5720,99 @@ mod tests {
 
             handle_child_output(r, &output);
         }
+
+        #[test]
+        fn test_windows_guest_cpu_hotplug() {
+            let resize_to_num = 6;
+
+            let mut windows = WindowsDiskConfig::new(WINDOWS_IMAGE_NAME.to_string());
+            let guest = Guest::new(&mut windows);
+
+            let tmp_dir = TempDir::new_with_prefix("/tmp/ch").unwrap();
+            let mut workload_path = dirs::home_dir().unwrap();
+            workload_path.push("workloads");
+
+            let mut ovmf_path = workload_path.clone();
+            ovmf_path.push(OVMF_NAME);
+
+            let mut osdisk_path = workload_path;
+            osdisk_path.push(WINDOWS_IMAGE_NAME.to_string());
+
+            let api_socket = temp_api_path(&tmp_dir);
+
+            let mut child = GuestCommand::new(&guest)
+                .args(&["--api-socket", &api_socket])
+                .args(&["--cpus", "boot=2,max=8,kvm_hyperv=on"])
+                .args(&["--memory", "size=4G"])
+                .args(&["--kernel", ovmf_path.to_str().unwrap()])
+                .default_disks()
+                .args(&["--serial", "tty"])
+                .args(&["--console", "off"])
+                .args(&["--net", "tap="])
+                .capture_output()
+                .spawn()
+                .unwrap();
+
+            // Wait to make sure Windows boots up
+            thread::sleep(std::time::Duration::new(60, 0));
+
+            let r = std::panic::catch_unwind(|| {
+                let auth = windows_auth();
+
+                // Issue hotadd CPUs
+                assert!(remote_command2(
+                    &api_socket,
+                    "resize",
+                    &["--cpus", format!("{}", resize_to_num).as_str()]
+                ));
+
+                // Wait to make sure CPUs are added
+                thread::sleep(std::time::Duration::new(3, 0));
+
+                // Check the guest sees the correct number
+                assert_eq!(
+                    ssh_command_ip_with_auth(
+                        "powershell -Command \"(Get-CimInstance win32_computersystem).NumberOfLogicalProcessors\"",
+                        &auth,
+                        "192.168.249.2",
+                        DEFAULT_SSH_RETRIES,
+                        DEFAULT_SSH_TIMEOUT,
+                    )
+                    .unwrap()
+                    .trim()
+                    .parse::<u8>()
+                    .unwrap_or(0),
+                    resize_to_num
+                );
```

Comment:
Factorize this into a dedicated function `get_cpu_count_windows()`. This will help with readability.

Reply:
Done, thanks.